src/lib/cloud/social/provider.ts

```diff
@@ -179,6 +168,13 @@ export class CloudSocialProvider {
   // SSH connections, keyed by host.
   private clients_: { [host: string]: Promise<Connection> } = {};
 
+  // Map from host to whether it is online.  Hosts not in the map are assumed
+  // to be offline.
+  private onlineHosts_: { [host: string]: boolean } = {};
```

Comment:
OK, I agree that would be a better design but we can revisit it in a future pull request